src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -11,18 +11,43 @@ public static class PageDirectiveFeature
     {
         public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
```

Comment:
I feel like this function is kind of serving double duty to determing if it's a RazorPage (the bool return value) and extract the template (the out value). If that was intentional why, and if not might it be clearer with two functions?